builtin/providers/aws/resource_aws_vpc_endpoint_route_table_association.go

```diff
@@ -0,0 +1,189 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsVpcEndpointRouteTableAssociation() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsVPCEndpointRouteTableAssociationCreate,
+		Read:   resourceAwsVPCEndpointRouteTableAssociationRead,
+		// "resource aws_vpc_endpoint_route_table_association: All fields are ForceNew or Computed w/out Optional, Update is superfluous"
+		// Update: resourceAwsVPCEndpointRouteTableAssociationUpdate,
+		Delete: resourceAwsVPCEndpointRouteTableAssociationDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vpc_endpoint_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"route_table_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+func resourceAwsVPCEndpointRouteTableAssociationCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).ec2conn
+	endpointId := d.Get("vpc_endpoint_id").(string)
+	rtId := d.Get("route_table_id").(string)
+
+	_, err := findResourceVPCEndpoint(conn, endpointId)
+	if err != nil {
+		return err
+	}
+
+	log.Printf(
+		"[INFO] Creating VPC Endpoint/Route Table association: %s => %s",
+		endpointId, rtId)
+
+	input := &ec2.ModifyVpcEndpointInput{
+		VpcEndpointId:    aws.String(endpointId),
+		AddRouteTableIds: aws.StringSlice([]string{rtId}),
+	}
+
+	_, err = conn.ModifyVpcEndpoint(input)
+	if err != nil {
+		return fmt.Errorf("Error creating VPC Endpoint/Route Table association: %s", err.Error())
+	}
+
```

Comment:
I believe generally you should be calling the `d.SetId` here to populate the state, the fact that you call `Read` immediately makes it a bit less necessary but is the pattern I've seen around the code base.